### ISSUE TITLE
module cron set_job for entries with 0/xx values and/or piped to > /dev/null 2>&1

### ISSUE THREAD

**Issue by kirk-hammond** [saltstack/salt#41815]
### Description of Issue/Question
I was attempting to comment out several cronjobs that were entered as show here:
_*/20 * * * * /root/scripts/agentMQ.sh > /dev/null 2>&1_

I am not sure if this is caused by the 0/20 or by the pipe to dev null, but I was unable to get salt cron module to update the cron entry for root to be commented out and managed by salt.

Dev box did not have 0/20 or pipe to /dev/null and salt managed it as expected. The cron job was updated to be below the managed by salt line and commented out with DISABLED in comment.

### Setup
Using module, this was the commands attempted.
salt 'target_host*' cron.set_job root '0/20' '*' '*' '*' '*' '/root/scripts/script.sh > /dev/null 2>&1' commented=True

### Steps to Reproduce Issue
Add line into root crontab file with 0/xx and pipe to dev/null.
Attempt to comment it out with cron.set_job commented=True

### Versions Report
**MASTER**
salt --versions-report
Salt Version:
           Salt: 2016.11.5
 
Dependency Versions:
           cffi: 1.9.1
       cherrypy: 3.2.2
       dateutil: 2.6.0
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.3
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Sep  1 2016, 22:14:00)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist:   
        machine: x86_64
        release: 4.9.20-10.30.amzn1.x86_64
         system: Linux
        version: Not Installed

** MINIONS ***
Minion-1
Salt Version:
               Salt: 2016.11.3
     
    Dependency Versions:
               cffi: Not Installed
           cherrypy: Not Installed
           dateutil: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.8.1
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.5
       mysql-python: Not Installed
          pycparser: Not Installed
           pycrypto: 2.6.1
             pygit2: Not Installed
             Python: 2.6.8 (unknown, Jul 18 2016, 16:53:33)
       python-gnupg: Not Installed
             PyYAML: 3.08
              PyZMQ: 14.5.0
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.2.1
                ZMQ: 4.0.5
     
    System Versions:
               dist: redhat 5.11 Tikanga
            machine: x86_64
            release: 2.6.18-419.el5
             system: Linux
            version: Red Hat Enterprise Linux Server 5.11 Tikanga
Minion-2
  Salt Version:
               Salt: 2016.11.3
     
    Dependency Versions:
               cffi: Not Installed
           cherrypy: Not Installed
           dateutil: 1.4.1
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.8.1
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: 0.20.2
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.6
       mysql-python: Not Installed
          pycparser: Not Installed
           pycrypto: 2.6.1
             pygit2: Not Installed
             Python: 2.6.6 (r266:84292, Aug  9 2016, 06:11:56)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 14.5.0
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.2.1
                ZMQ: 4.0.5
    
    System Versions:
               dist: redhat 6.6 Santiago
            machine: x86_64
            release: 2.6.32-696.1.1.el6.x86_64
             system: Linux
            version: Red Hat Enterprise Linux Server 6.6 Santiago

Minion-3
    Salt Version:
               Salt: 2016.11.3
     
    Dependency Versions:
               cffi: Not Installed
           cherrypy: Not Installed
           dateutil: 1.4.1
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.8.1
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: 0.20.2
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.6
       mysql-python: Not Installed
          pycparser: Not Installed
           pycrypto: 2.6.1
             pygit2: Not Installed
             Python: 2.6.6 (r266:84292, Aug  9 2016, 06:11:56)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 14.5.0
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.2.1
                ZMQ: 4.0.5
     
    System Versions:
               dist: redhat 6.6 Santiago
            machine: x86_64
            release: 2.6.32-696.1.1.el6.x86_64
             system: Linux
            version: Red Hat Enterprise Linux Server 6.6 Santiago

**Comment by garethgreenaway** [comment]
@kirk-hammond Thanks for the report.  Some additional questions.  Were the jobs in question previously added to the crontab for the root user by Salt or were they manually added?  Also, can you provide the output for the cases where the module does not perform the desired action?  Thanks!

**Comment by kirk-hammond** [comment]
All of the jobs in question were previously added to the root users crontab. If I deleted the old job, created a new one using salt.cron with the exact same cron data it was properly managed and could be set to disabled or enabled from that point without issue.

These are legacy systems and I did not capture the output unfortunately. I updated the crons to stop them as they are no longer needed and enabling to test is not an option. I plan to re-create this in a virtual lab in the near future to confirm the issue can be produced on non-legacy (read old n' busted) hosts.
